tests/src/Interop/PInvoke/IEnumerator/IEnumeratorTest.cs

```diff
@@ -0,0 +1,89 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.CustomMarshalers;
+using TestLibrary;
+
+using Console = Internal.Console;
+
+namespace PInvokeTests
+{
+    static class IEnumeratorNative
+    {
+        [DllImport(nameof(IEnumeratorNative), PreserveSig = false)]
+        public static extern IEnumerator GetIntegerEnumerator(
+            int start,
+            int count);
+
+        [DllImport(nameof(IEnumeratorNative), PreserveSig = false)]
+        [return: MarshalAs(UnmanagedType.CustomMarshaler, MarshalType = "System.Runtime.InteropServices.CustomMarshalers.EnumerableToDispatchMarshaler, System.Private.CoreLib, Culture=neutral, PublicKeyToken=B77A5C561934E089, Version=4.0.0.0")]
```

Comment:
Given the other conversation, I assume that we expect customer code to do this as well. This should be changed to use the type from the public location. It would  be useful to add TODO/comment about it.

Reply:
Until we expose these types in corefx, that is correct. However, as we talked about before, we didn't find any customers (or a number so small that it showed up as 0% on apiport) directly using this marshaler. Therefore, I don't think it's a pressing issue to expose these marshalers via corefx. The only reason I'm doing what I'm doing here is so that we can get some tests for them.

Reply:
ok